lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -107,7 +108,16 @@ public synchronized void onRestart() {
 
     driverRuntimeRestartManager.informAboutEvaluatorFailures(getFailedEvaluators());
 
-    // TODO[REEF-560]: Call onDriverRestartCompleted() on a Timer.
+    if (driverRestartEvaluatorRecoveryMillis != Integer.MAX_VALUE) {
+      // Don't use Clock here because if there is an event scheduled, the driver will not be idle, even if
+      // driver restart has already completed, and we cannot cancel the event.
+      new Timer().schedule(new TimerTask() {
+        @Override
+        public void run() {
+          onDriverRestartCompleted();
```

Comment:
The semantics here is that we *always* generate the restart completed event. What else is signified by this event to the user? Example: What happens if we get an evaluator checkin after this time? Do we declare them dead?

Also: Should we tell the app that we fired this event via a timeout as opposed to because we have all required information?

Lastly: Will this event also fire once we've heard from all evaluators or only after the timer? is there a situation where it fires twice?